subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -920,31 +926,6 @@ public TestFramework testFramework(@Nullable Closure testFrameworkConfigure) {
             useJUnit(testFrameworkConfigure);
         }
 
-        // To maintain backwards compatibility with builds that may configure the test framework
-        // multiple times for a single task--either switching between frameworks or overwriting
-        // the existing configuration for a test framework--we need to keep track if the user has
-        // explicitly set a test framework
-        // With test suites, users should never need to call the useXXX methods, so we can warn if
-        // them from doing something like this (for now, in order to preserve existing builds).
-        // This behavior should be restored to fail fast once again with the next major version.
-        //
-        // testTask.configure {
-        //      options {
-        //          /* configure JUnit Platform */
-        //      }
-        // }
-        // testTask.configure {
-        //      useJUnit()
-        //      options {
-        //          /* configure JUnit */
-        //      }
-        // }
-
-        // TODO: Test Framework Selection - Restore this to re-enable fail-fast behavior for Gradle 8
```

Comment:
Why are we not restoring this behavior?

Reply:
While that also works to create the same behavior, the error message it produces is less than clear.  Decided to go with a custom message in the `useTestFramework` method.

Reply:
In that case we should be able to remove the `userHasConfiguredTestFramework` variable, since it is now only set and never read.